### ISSUE TITLE
ISOTPSoftSocket keeps process alive

### ISSUE THREAD

**Issue by akorb** [secdev/scapy#1753]
Simple script to reprodue:

```python
from scapy.contrib.isotp import *

socket = ISOTPSocket('vcan0', sid=0x7E0, did=0x7E8)
```

The `socket.close()` command is missing. This results in `ISOTPSoftSocket.rx_thread` not beeing closed and keeping the process alive.
Especially annoying if you use it with the interactive shell and calling `exit()` never returns.
Furthermore, the `ISOTPNativeSocket` implementation exits properly since it does not create an own thread. But they should be interchangeable and thus have the same behavior.

A simple solution would be to set the rx_thread as a daemon:
```python
self.rx_thread = CANReceiverThread(can_socket, can_on_recv)
self.rx_thread.daemon = True
self.rx_thread.start()
```
But I cannot judge if this is the final solution.

@epozzobon 
@polybassa

**Comment by guedou** [comment]
@polybassa do you consider this issue as closed?

**Comment by polybassa** [comment]
Yes. We are currently working on a better solution, but in most cases, a background thread is required. A user of a ISOTPSoftSocket have to care about closing a thread, anyway.
